### DISCUSSION TITLE
Getting this: BUSY Redis is busy running a script. You can only call SCRIPT KILL or SHUTDOWN

### DISCUSSION THREAD

**Discussion by raghavFam** [valkey-io/valkey#1404] [Q&A]
We are running Valkey 8.0.1 on AWS Elasticache. We are using it to cache our database queries using django-cacheops. For a few days it was working fine at scale but suddenly we started getting multiple errors as shared in the title. Any possible reason why this is happening?
We checked Stackoverflow: https://serverfault.com/questions/999431/how-to-fix-the-redis-is-busy-running-a-script-error
There it was mentioned that since Redis is single threaded this might occur, but as far as I know based on docs, valkey 8.0.1 is multithreaded right? 

Before Valkey we were using Redis 7.4.0.

It would be great if someone can share their insights on the matter.

**Comment by sandeeeep** [comment]
@madolson any help here is much appreciated!

**Reply by madolson** [reply]
Hi! Even though Valkey 8.0.1 is multi-threaded, LUA is all still run synchronously today, so it blocks other operations. Do you have any more information on the script being run that is taking so long?
